library/ecdsa.c

```diff
@@ -286,10 +286,71 @@ int mbedtls_ecdsa_verify( mbedtls_ecp_group *grp,
 }
 #endif /* MBEDTLS_ECDSA_VERIFY_ALT */
 
+/*
+ * Convert a signature to a raw concatenation of {r, s}
+ */
+int mbedtls_ecdsa_signature_to_raw( const unsigned char *sig,
+                            size_t ssize, uint16_t byte_len,
+                            unsigned char *buf, size_t* slen )
+{
+    int ret;
+    unsigned char *p = (unsigned char *) sig;
+    const unsigned char *end = sig + ssize;
+    size_t len;
+    mbedtls_mpi r, s;
+
+    if( 2 * byte_len > ssize )
+    {
+        return MBEDTLS_ERR_ECP_BAD_INPUT_DATA;
+    }
+
+    mbedtls_mpi_init( &r );
+    mbedtls_mpi_init( &s );
```

Comment:
As mentioned above, while conceptually the clearest, I think we should aim to avoid costly dynamic allocation for such a simple conversion task.

Reply:
I agree. IIRC I originally did it that way because avoiding dynamic allocation would have required extending the mpi module.

Reply:
Done for mbedtls_ecdsa_signature_to_raw, please re-review.